### PR TITLE
Issue #2231, refactored contact_page file

### PR DESCRIPTION
**Pull request by michael-volynets**
Issue #2231,
Time Spent: 30m

### CODE REVIEW

File: extension/chrome/settings/modules/contact_page.ts [hunk 1]
```diff
@@ -6,76 +6,102 @@ import { Catch } from '../../../js/common/platform/catch.js';
...
           Xss.sanitizeReplace('#select_photo', Ui.e('span', { text: file.name }));
         }
       });
     } else {
-      S.cached('management_account').text(result.email).parent().removeClass('display_none');
-      Xss.sanitizeRender(S.cached('status'), 'Your contact page is currently <b class="bad">disabled</b>. <a href="#" class="action_enable">Enable contact page</a>');
-      S.now('action_enable').click(Ui.event.prevent('double', enableContactPage));
+      this.S.cached('management_account').text(result.email).parent().removeClass('display_none');
+      Xss.sanitizeRender(this.S.cached('status'), 'Your contact page is currently <b class="bad">disabled</b>. <a href="#" class="action_enable">Enable contact page</a>');
+      this.S.now('action_enable').click(Ui.event.prevent('double', () => this.enableContactPage()));
```
Comment (tomholub) [inline extension/chrome/settings/modules/contact_page.ts]: Please use `this.setHandlerPrevent`

Comment (tomholub) [reply extension/chrome/settings/modules/contact_page.ts]: and then `(this.enableContactPage)` directly

Comment (michael-volynets) [reply extension/chrome/settings/modules/contact_page.ts]: Sorry, I missed that!
Thanks!

**Comment by michael-volynets** [comment]
> Do you have a decent way to test this page?

I tested if the page updates data.
Or I might have not exactly understood what you mean.

File: extension/chrome/settings/modules/contact_page.ts [hunk 2]
```diff
@@ -91,27 +91,25 @@ View.run(class ContactPageView extends View {
...
-        await Ui.modal.error('Failed to enable your Contact Page. Please try again');
-      }
-      await Ui.time.sleep(100);
-      window.location.reload();
-    } catch (e) {
-      Catch.reportErr(e);
-      await Ui.modal.error(`Failed to create account, possibly a network issue. Please try again.\n\n${String(e)}`);
-      await Ui.time.sleep(100);
-      window.location.reload();
+      this.S.now('action_enable').click(this.setHandlerPrevent('double', async () => {
```
Comment (tomholub) [inline extension/chrome/settings/modules/contact_page.ts]: why not continue to have it as a method, eg `this.S.now('action_enable').click(this.setHandlerPrevent('double', this.enableContactPage))` ?

Comment (michael-volynets) [reply extension/chrome/settings/modules/contact_page.ts]: Sorry, I didn't understand exactly what you meant in the previous comment. 
> and then (this.enableContactPage) directly

I thought you want me to remove the method and paste it directly in the `this.setHandlerPrevent('double'`
My bad. I will restore the code.

Comment (tomholub) [reply extension/chrome/settings/modules/contact_page.ts]: Sorry, maybe I wasn't clear.

File: extension/chrome/settings/modules/contact_page.ts [hunk 3]
```diff
@@ -91,25 +91,7 @@ View.run(class ContactPageView extends View {
...
-          await Ui.time.sleep(100);
-          window.location.reload();
-        } catch (e) {
-          Catch.reportErr(e);
-          await Ui.modal.error(`Failed to create account, possibly a network issue. Please try again.\n\n${String(e)}`);
-          await Ui.time.sleep(100);
-          window.location.reload();
-        }
-      }));
+      this.S.now('action_enable').click(this.setHandlerPrevent('double', () => this.enableContactPage()));
```
Comment (tomholub) [inline extension/chrome/settings/modules/contact_page.ts]: Here you can simplify to
```
       this.S.now('action_enable').click(this.setHandlerPrevent('double', this.enableContactPage));
```

Comment (tomholub) [reply extension/chrome/settings/modules/contact_page.ts]: I'm taking this back, all handlers will use arrow functions for safety and consistency, the way you did it above. See #2253

Comment (michael-volynets) [reply extension/chrome/settings/modules/contact_page.ts]: Okay, I will use only arrow functions for handlers,
Thanks!

**Pull request merged**